frontend/src/components/TypeaheadSelect.tsx

```diff
@@ -400,33 +406,123 @@ const TypeaheadSelect: React.FunctionComponent<TypeaheadSelectProps> = ({
+              onChange={onTextInputChange}
+              onKeyDown={onInputKeyDown}
+              autoComplete="off"
+              innerRef={textInputRef}
+              placeholder={placeholder}
+              {...(activeItemId && { 'aria-activedescendant': activeItemId })}
+              role="combobox"
+              isExpanded={isOpen}
+              aria-controls="select-typeahead-listbox"
             />
-          </TextInputGroupUtilities>
-        ) : null}
+          </FlexItem>
+          {selected && selected.selectedLabel && <FlexItem>{selected.selectedLabel}</FlexItem>}
+          {(isFiltering && filterValue) || (allowClear && selected) ? (
+            <FlexItem>
+              <TextInputGroupUtilities>
+                <Button
+                  icon={<TimesIcon aria-hidden />}
+                  variant="plain"
+                  onClick={onClearButtonClick}
+                  aria-label="Clear input value"
+                />
+              </TextInputGroupUtilities>
+            </FlexItem>
+          ) : null}
+        </Flex>
       </TextInputGroup>
     </MenuToggle>
   );
 
+  const groupedSelections = React.useMemo(() => {
+    const group: Record<string, { groupLabel: React.ReactNode; options: TypeaheadSelectOption[] }> =
+      {};
+    const noGroup: TypeaheadSelectOption[] = [];
+
+    filteredSelections.forEach((option) => {
+      if (option.group) {
+        const { groupName } = option.group;
+        const { groupLabel } = option.group;
+
+        if (groupName in group) {
+          group[groupName].options.push(option);
+        } else {
+          group[groupName] = {
+            groupLabel,
+            options: [option],
+          };
+        }
+      } else {
+        noGroup.push(option);
+      }
+    });
+
+    return { group, noGroup };
+  }, [filteredSelections]);
+
+  const tSelectOption = (option: TypeaheadSelectOption, index: number) => {
+    const { content, value, dropdownLabel, ...optionProps } = option;
+    return (
+      <SelectOption
+        key={value}
+        value={value}
+        isFocused={focusedItemIndex === index}
+        {...optionProps}
+      >
+        {dropdownLabel ? (
+          <Flex>
+            <FlexItem>{content}</FlexItem>
+            <FlexItem>{dropdownLabel}</FlexItem>
+          </Flex>
+        ) : (
+          content
+        )}
+      </SelectOption>
+    );
+  };
+
+  const tGroupOption = (
+    groupName: string,
+    group: { groupLabel: React.ReactNode; options: TypeaheadSelectOption[] },
+    startingIndex: number,
+  ): { node: React.ReactNode; nextIndex: number } => {
+    let index = startingIndex;
+    return {
+      node: (
+        <SelectGroup
+          key={groupName}
+          label={group.groupLabel}
+          data-testid={`typeahead-group-${groupName.toLowerCase()}`}
+        >
+          {group.options.map((opt) => tSelectOption(opt, index++))}
+        </SelectGroup>
+      ),
+      nextIndex: index,
+    };
+  };
```

Comment:
We should check if the group is the last group and render a `<Divider />` below if it's not the last one